enterprise/api_client/enterprise_catalog.py

```diff
@@ -143,37 +143,53 @@ def delete_enterprise_catalog(self, catalog_uuid):
 
     @JwtLmsApiClient.refresh_token
-    def get_content_metadata(self, enterprise_customer, enterprise_catalogs=None):
+    def get_content_metadata(self, enterprise_customer_catalogs, catalogs_last_modified=None):
         """
         Return all content metadata contained in the catalogs associated with the EnterpriseCustomer.
 
         Arguments:
-            enterprise_customer (EnterpriseCustomer): The EnterpriseCustomer to return content metadata for.
-            enterprise_catalogs (EnterpriseCustomerCatalog): Optional list of EnterpriseCustomerCatalog objects.
+            enterprise_customer_catalogs (List of dicts): list of enterprise catalog data.
+                Example:
+                    [
+                        {
+                            uuid: <catalog UUID>,
+                            title: <catalog title>,
+                                ...
+                        }, {
+                            uuid: <catalog UUID>,
+                            title: <catalog title>,
+                                ...
+                    ]}
+            catalogs_last_modified (Dict): Optional mapping between catalog UUIDs and the last modified at time of the
+                catalog
 
         Returns:
-            list: List of dicts containing content metadata.
+            list(content_metadata): List of content metadata items dicts,
+
+            content_catalog_last_modified: Mapping of content keys to the last modified at time of their associated
+                catalog
         """
         content_metadata = OrderedDict()
-        enterprise_customer_catalogs = enterprise_catalogs or enterprise_customer.enterprise_customer_catalogs.all()
-
+        content_catalog_last_modified = {}
         for enterprise_customer_catalog in enterprise_customer_catalogs:
-            catalog_uuid = enterprise_customer_catalog.uuid
+            catalog_uuid = enterprise_customer_catalog.get('uuid')
             endpoint = getattr(self.client, self.GET_CONTENT_METADATA_ENDPOINT.format(catalog_uuid))
             query = {'page_size': self.GET_CONTENT_METADATA_PAGE_SIZE}
             try:
                 response = endpoint.get(**query)
                 for item in utils.traverse_pagination(response, endpoint):
                     content_id = utils.get_content_metadata_item_id(item)
                     content_metadata[content_id] = item
+                    if catalogs_last_modified:
+                        content_catalog_last_modified[content_id] = catalogs_last_modified[catalog_uuid]
             except (SlumberBaseException, ConnectionError, Timeout) as exc:
                 LOGGER.exception(
                     'Failed to get content metadata for Catalog [%s] in enterprise-catalog due to: [%s]',
                     catalog_uuid, str(exc)
                 )
                 raise
 
-        return list(content_metadata.values())
+        return list(content_metadata.values()), content_catalog_last_modified
```

Comment:
The other thing you chould choose to do for better backwards compatibility is to not return the 2nd parameter if catalogs_last_modified is passed in as None.